### ISSUE TITLE
cf_protection unsupported option for target AArch64

### ISSUE THREAD

**Issue by lorenzo-carpentieri** [intel/llvm#14458]
### Describe the bug

I'm compiling DPC++ ([commit](https://github.com/intel/llvm/commit/36c88dcab2be8f04198e57b93d702c806616c4c9)) on OrangePi 5 Plus equipped with a Rockchip RK3588 8-core 64-bit processor, quad-core A76+quad-core A55  with an integrated ARM Mali-G610 GPU.
The DPC++ compilation process ends with:
```
error: option 'cf-protection=return' cannot be specified on this target
error: option 'cf-protection=branch' cannot be specified on this target
```

### To reproduce

1. Configure command: `python3 ./configure.py -t Release -o BUILD_PATH --cmake-gen "Unix Makefiles"  --host-target="AArch64`
2. Compile command: `python3 ./compile.py -o  BUILD_PATH -j 8`
4. Error: **error: option 'cf-protection=return' cannot be specified on this target**

### Environment

- OS: Ubuntu 22.04.3 LTS
- Target device and vendor:  Rockchip RK3588 8-core 64-bit processor, quad-core A76+quad-core A55  with an integrated ARM Mali-G610 GPU
- DPC++ version: ([commit](https://github.com/intel/llvm/commit/36c88dcab2be8f04198e57b93d702c806616c4c9))
-  gcc version 11.4

**Comment by coldav** [comment]
Note related fix https://github.com/intel/llvm/commit/00915ba7d6999a7d4c3cb62cf60fe8117310346f

**Comment by dm-vodopyanov** [comment]
@coldav thanks for the fix. Am I correct that the issue is resolved now?
@lorenzo-carpentieri hello, thanks for the report, did https://github.com/intel/llvm/commit/00915ba7d6999a7d4c3cb62cf60fe8117310346f help you?

**Comment by coldav** [comment]
> @coldav thanks for the fix. Am I correct that the issue is resolved now? @lorenzo-carpentieri hello, thanks for the report, did [00915ba](https://github.com/intel/llvm/commit/00915ba7d6999a7d4c3cb62cf60fe8117310346f) help you?

I don't know if it fixes your issue - but it definitely seems related.

**Comment by lorenzo-carpentieri** [comment]
Thank you for your quick reply. 
The fix solved the problem and I compiled DPC++ correctly. 
However, the sycl-ls --verbose command generates a SYCL exception.
I also tried two implementations of a vector addition (1) allocating the data with USM malloc_device and  (2) with buffer/accessor.
(1) fail with USM Device does not support USM device allocations! After some investigation, I think is related to the fact that orange PI Plus does not have support for FINE_GRAIN svmAlloc. Am I right?

(2) fail with  Native API failed. Native API returns: -30 (PI_ERROR_INVALID_VALUE) -30 (PI_ERROR_INVALID_VALUE). With buffer/ accessor I think the problem is the same. 

![image](https://github.com/intel/llvm/assets/57230554/93e164b3-2e00-4cab-89ae-455b543d89dc)
